torch/_inductor/codegen/triton.py

```diff
@@ -1974,7 +1974,10 @@ def load(self, name: str, index: sympy.Expr):
                 line = f"tl.load({var} + ({original_index}))"
                 append_broadcast = indexing.expand_str
             else:
-                line = f"tl.load({var} + ({indexing.index_str}), {indexing.mask_str}{ep}{other})"
+                if torch.version.hip is not None and indexing.mask_str != "None":
+                    line = f"tl.load({var} + ({indexing.index_str}), ({indexing.mask_str}).to(tl.int1){ep}{other})"
```

Comment:
NIT: It would be better to convert to boolean in `ops.masked` rather than at the point where the mask is used.